api/s2n.h

```diff
@@ -302,6 +302,11 @@ extern int s2n_config_add_ticket_crypto_key(struct s2n_config *config,
                                             uint8_t *key, uint32_t key_len,
                                             uint64_t intro_time_in_seconds_from_epoch);
 
+S2N_API
+extern int s2n_config_set_ctx(struct s2n_config *config, void *ctx);
+S2N_API
+extern void *s2n_config_get_ctx(const struct s2n_config *config);
+
```

Comment:
I know you're matching the existing connection version of this API, but do we want to risk deviating from our "must return int" rule? The method is simple for now, but we can't change the return value later.

Reply:
what would the new api look like:
```
extern int s2n_config_get_ctx(const struct s2n_config *config, void *ctx);
```

Btw not opposed to this but would like to hear if there a case for the current signature.

Reply:
It's a valid question... Do you stay consistent with what's there or go with the newer convention? On the one hand, I don't think this will ever be anything but `config ? config->ctx : NULL` so having to deal with an out parameter isn't the best. But on the other hand it would be nice to be strict about return values for any new APIs.